Interpreter doesn't find libiconv on master on Mac

```
$ bin/crystal i
Using compiled compiler at .build/crystal
cannot find -liconv
Linker arguments: -L/opt/homebrew/lib -L/opt/homebrew/Cellar/libevent/2.1.12/lib -levent -liconv
Search path: /usr/lib:/usr/local/lib:/opt/homebrew/lib:/opt/homebrew/Cellar/libevent/2.1.12/lib (Crystal::Loader::LoadError)
  from src/compiler/crystal/loader/unix.cr:96:31 in 'load_library'
  from src/compiler/crystal/loader.cr:42:7 in 'new'
  from src/compiler/crystal/loader/unix.cr:60:7 in 'parse'
```

--- a/src/compiler/crystal/loader.cr
+++ b/src/compiler/crystal/loader.cr
@@ -89,7 +89,7 @@ class Crystal::Loader
 
     @search_paths.each do |directory|
       library_path = File.join(directory, Loader.library_filename(libname))
-      return true if File.file?(library_path) && load_file?(library_path)
+      return true if load_file?(library_path)
     end
 
     false

